From 53c472b6022288d480f14730c8532bea416f845e Mon Sep 17 00:00:00 2001
From: John Zhuge <jzhuge@cloudera.com>
Date: Mon, 25 Sep 2017 18:35:35 -0700
Subject: [PATCH 2645/2848] HDFS-12424. Datatable sorting on the Datanode
 Information page in the Namenode UI is broken.
 Contributed by Shawna Martell.

(cherry picked from commit 78bdf10ae40300a5f2f3b95491d280802edc1df2)

Conflicts:
	hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/hdfs/dfshealth.js

Change-Id: I7ede1b3b6d19c54f2d3fbc1689482a693ea541aa
---
 .../hadoop-hdfs/src/main/webapps/hdfs/dfshealth.js |   10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/hdfs/dfshealth.js b/hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/hdfs/dfshealth.js
index 607dd64..8dde227 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/hdfs/dfshealth.js
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/hdfs/dfshealth.js
@@ -321,11 +321,11 @@
             'lengthMenu': [ [25, 50, 100, -1], [25, 50, 100, "All"] ],
             'columns': [
               { 'orderDataType': 'ng-value', 'searchable': true },
-              { 'orderDataType': 'ng-value', 'type': 'numeric' },
-              { 'orderDataType': 'ng-value', 'type': 'numeric' },
-              { 'orderData': 3, 'type': 'numeric' },
-              { 'orderDataType': 'ng-value', 'type': 'numeric'},
-              { 'orderData': 5 }
+              { 'orderDataType': 'ng-value', 'type': 'num' },
+              { 'orderDataType': 'ng-value', 'type': 'num' },
+              { 'type': 'num' },
+              { 'orderDataType': 'ng-value', 'type': 'num'},
+              { 'type': 'string' }
             ]});
           renderHistogram(data);
           $('#ui-tabs a[href="#tab-datanode"]').tab('show');
-- 
1.7.9.5

